### PR TITLE
[FIX] Profile fields automatically reset

### LINKED ISSUE #1496: The field is resetting
- Your Rocket.Chat Experimental app version: 4.2.0
- Your Rocket.Chat server version: 2.4.0-rc.0
- Device (or Simulator) you're running with: Android & iOS

For the profile part, after I keyed-in value to the field, then it seems like is resetting automatically in awhile. 
And also for the image, i try to select image from picker and debug using Reactotron it showed there is value in the avatar object after selected, but it also seems like resetting and became empty right after.

reset field

### PR DESCRIPTION
**Pull request by ssbing99**
@RocketChat/ReactNative

Closes #1496

### CODE REVIEW

File: app/views/ProfileView/index.js [hunk 1]
```diff
@@ -81,7 +82,7 @@ class ProfileView extends React.Component {
 
 	componentWillReceiveProps(nextProps) {
 		const { user } = this.props;
-		if (user !== nextProps.user) {
+		if (isEqual(user, nextProps.user)) {
```
Comment (djorkaeffalexandre) [inline app/views/ProfileView/index.js]: You can use `equal` from `deep-equal` and not need to import `lodash/isEqual` :)

Comment (ssbing99) [reply app/views/ProfileView/index.js]: Oops , sorry my bad, will resubmit again

Comment (ssbing99) [reply app/views/ProfileView/index.js]: I have pushed another commit

File: app/views/ProfileView/index.js [hunk 2]
```diff
@@ -81,7 +81,7 @@ class ProfileView extends React.Component {
 
 	componentWillReceiveProps(nextProps) {
 		const { user } = this.props;
-		if (user !== nextProps.user) {
+		if (equal(user, nextProps.user)) {
```
Comment (djorkaeffalexandre) [inline app/views/ProfileView/index.js]: `if (!equal(user, nextProps.user))`

Comment (ssbing99) [reply app/views/ProfileView/index.js]: argh .... have pushed another commit. Thank you

**Pull request merged**